src/test/java/io/confluent/examples/streams/IntegrationTestUtils.java

```diff
@@ -94,13 +96,13 @@
   public static <K, V> List<KeyValue<K, V>> readKeyValues(final String topic, final Properties consumerConfig, final int maxMessages) {
     final KafkaConsumer<K, V> consumer = new KafkaConsumer<>(consumerConfig);
     consumer.subscribe(Collections.singletonList(topic));
-    final int pollIntervalMs = 100;
-    final int maxTotalPollTimeMs = 2000;
+    final Duration pollInterval = Duration.ofMillis(100L);
+    final int maxTotalPollTimeMs = 10000;
     int totalPollTimeMs = 0;
     final List<KeyValue<K, V>> consumedValues = new ArrayList<>();
     while (totalPollTimeMs < maxTotalPollTimeMs && continueConsuming(consumedValues.size(), maxMessages)) {
-      totalPollTimeMs += pollIntervalMs;
-      final ConsumerRecords<K, V> records = consumer.poll(pollIntervalMs);
+      totalPollTimeMs += pollInterval.toMillis();
```

Comment:
would it be better to just increase this by the actual time spent in `poll`?

Reply:
We can do this -- even if it does not change much, because we stay about 100ms in `poll(Duration)` per call anyway.

Reply:
I'll leave it to you.

Since the timeout for the test is defined in terms of the time spent, it might make more sense to use the time spent. Otherwise it's effectively just counting the _number_ of polls, but in a roundabout way, where the max number of polls is expressed as `maxTotalPollTimeMs / pollInterval.toMillis` and each time, we count up by incrementing `pollInterval.toMillis` amount.

It's a little janky as-is, but not a major concern for me.

Reply:
Yes. Updated as request. Does not make a big difference, but as you said, it's more precise to actually measure it, instead of counting poll-calls.